roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/util/ConfigParserTest.java

```diff
@@ -0,0 +1,215 @@
+        URL path = getClass().getClassLoader().getResource("test/roundabout.xml");
+        assertNotNull(path);
+
+        configParser = new ConfigParser(path.getPath());
+        roundAboutConfig = configParser.loadConfig();
+    }
+
+    @Test
+    public void configParserTest_parametersLoaded() {
+        assertNotNull("parameters attribute not null", roundAboutConfig.getParameters());
+        assertEquals("6 parameters loaded", 1, roundAboutConfig.getParameters().getParameter().size());
+    }
+
+    @Test
+    public void configParserTest_parametersDataLoaded() {
+        assertNotNull("parameter name loaded", roundAboutConfig.getParameters().getParameter().get(0).getName());
+        assertNotNull("parameter value loaded", roundAboutConfig.getParameters().getParameter().get(0).getValue());
+    }
+
+    @Test
+    public void configParserTest_roundaboutLoaded() {
+        assertNotNull("roundabout loaded", roundAboutConfig.getRoundabout());
+    }
+
+    @Test
+    public void configParserTest_roundaboutDataLoaded() {
+        assertNotNull("roundabout sections loaded", roundAboutConfig.getRoundabout().getSections());
+        assertEquals("roundabout correct name", "Kreisverkehr Dornbirn Süd", roundAboutConfig.getRoundabout().getName());
+        assertEquals("5 sections loaded in roundabout", 5, roundAboutConfig.getRoundabout().getSections().getSection().size());
+    }
+
+    @Test
+    public void configParserTest_sectionsDataLoaded() {
+        assertNotNull("sections has entry", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getEntry());
+        assertEquals("sections has 2 entries", 2, roundAboutConfig.getRoundabout().getSections().getSection().get(0).getEntry().size());
+        assertNotNull("sections has out", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getExit());
+        assertNotNull("sections has tracks", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getTrack());
+
+        assertNotNull("sections id", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getId());
+        assertNotNull("sections previous", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getPrevious());
+        assertNotNull("sections next", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getNext());
+    }
+
+    @Test
+    public void configParserTest_sectionDataLoaded() {
+        assertNotNull("section has tracks", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getTrack());
+        assertNotEquals("section has tracks", 0, roundAboutConfig.getRoundabout().getSections().getSection().get(0).getTrack().size());
+        assertNotNull("track has length", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getTrack().get(0).getLength());
+        assertNotNull("track has id", roundAboutConfig.getRoundabout().getSections().getSection().get(0).getTrack().get(0).getId());
```

Comment:
multiple asserts issue, see other comment

hint: create a variable like track = roundAboutConfig.getRoundabout().getSections().getSection().get(0).getTrack()
and use this variable in the test.. otherwise a change could be fatal and you would increase readability and maintainability

Reply:
same as above, if the section has no tracks, the track check would result in a null pointer exception